include/envoy/http/filter.h

```diff
@@ -418,6 +418,21 @@ class StreamDecoderFilterCallbacks : public virtual StreamFilterCallbacks {
   // Note that HttpConnectionManager sanitization will *not* be performed on the
   // recreated stream, as it is assumed that sanitization has already been done.
   virtual bool recreateStream() PURE;
+
+  /**
+   * Adds socket options to be applied to any connections used for upstream requests. Note that
+   * unique values for the options will likely lead to many connection pools being created. The
+   * added options are appended to any previously added. They will later be combined with any
```

Comment:
I'm probably missing something, but what does setting upstream socket options have to do with downstream socket options? Clarify?

Reply:
I replied below -- TL;DR: we probably shouldn't have the downstream comment.